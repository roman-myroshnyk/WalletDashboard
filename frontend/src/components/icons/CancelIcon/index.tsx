import { FC } from 'react';
import { motion } from 'framer-motion';
import { getToggleAnimations } from '@/animations/ToggleAnimations';

export interface IProps {
  label?: string;
  height?: number;
  width?: number;
  enterDelay?:number;
  exitDelay?:number;
  testid?: string;
 }

const CancelIcon:FC<IProps> = ({
  label, height, width, enterDelay, exitDelay, testid,
}) => {
  const variants = getToggleAnimations(enterDelay, exitDelay);
  return (
    <motion.svg
      xmlns="http://www.w3.org/2000/svg"
      className="h-6 w-6"
      aria-label={label}
      fill="none"
      width={width}
      height={height}
      viewBox="0 0 24 24"
      stroke="currentColor"
      strokeWidth={2}
      initial="exit"
      animate="enter"
      exit="exit"
      variants={variants.svg}
      data-testid={testid}
    >
      <motion.path
        variants={variants.path}
        strokeLinecap="round"
        strokeLinejoin="round"
        d="M6 18L18 6M6 12"
      />
      <motion.path
        variants={variants.path}
        strokeLinecap="round"
        strokeLinejoin="round"
        d="M6 6M6 6l12 12"
      />
    </motion.svg>
  );
};

CancelIcon.defaultProps = {
  label: 'cancelIcon',
  height: 24,
  width: 24,
  enterDelay: 0,
  exitDelay: 0,
  testid: 'cancelIcon',
};

export default CancelIcon;
