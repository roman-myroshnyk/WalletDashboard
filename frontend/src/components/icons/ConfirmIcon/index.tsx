import { FC } from 'react';
import { motion } from 'framer-motion';
import { getToggleAnimations } from '@/animations/ToggleAnimations';

export interface IProps {
  height?: number;
  width?: number;
  enterDelay?: number;
  exitDelay?: number;
 }

const ConfirmIcon:FC<IProps> = ({
  height, width, enterDelay, exitDelay,
}) => {
  const variants = getToggleAnimations(enterDelay, exitDelay);
  return (
    <motion.svg
      xmlns="http://www.w3.org/2000/svg"
      className="h-6 w-6"
      fill="none"
      width={width}
      height={height}
      viewBox="0 0 24 24"
      stroke="currentColor"
      strokeWidth={2}
      initial="exit"
      animate="enter"
      exit="exit"
      variants={variants.svg}
    >
      <motion.path
        variants={variants.path}
        strokeLinecap="round"
        strokeLinejoin="round"
        d="M5 13l4 4L19 7"
      />

    </motion.svg>
  );
};

ConfirmIcon.defaultProps = {
  height: 24,
  width: 24,
  enterDelay: 0,
  exitDelay: 0,
};

export default ConfirmIcon;
