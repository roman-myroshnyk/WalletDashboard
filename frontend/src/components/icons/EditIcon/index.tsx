import { FC } from 'react';
import { motion } from 'framer-motion';
import { getToggleAnimations } from '@/animations/ToggleAnimations';

export interface IProps {
  label?: string;
  height?: number;
  width?: number;
  enterDelay?: number;
  exitDelay?: number;
}

const EditIcon:FC<IProps> = ({
  label, height, width, enterDelay, exitDelay,
}) => {
  const variants = getToggleAnimations(enterDelay, exitDelay);
  return (
    <motion.svg
      initial="exit"
      animate="enter"
      exit="exit"
      variants={variants.svg}
      xmlns="http://www.w3.org/2000/svg"
      className="h-6 w-6"
      aria-label={label}
      fill="none"
      width={width}
      height={height}
      viewBox="0 0 24 24"
      stroke="currentColor"
      strokeWidth={2}
    >
      <motion.path
        variants={variants.path}
        strokeLinecap="round"
        strokeLinejoin="round"
        d="M11 5H6a2 2 0 00-2 2v11a2 2 0 002 2h11a2 2 0 002-2v-5m-1.414-9.414a2 2 0 112.828 2.828L11.828 15H9v-2.828l8.586-8.586z"
      />
    </motion.svg>
  );
};

EditIcon.defaultProps = {
  label: 'editIcon',
  height: 24,
  width: 24,
  enterDelay: 0,
  exitDelay: 0,
};

export default EditIcon;
