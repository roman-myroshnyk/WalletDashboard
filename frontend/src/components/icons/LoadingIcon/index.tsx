import { FC } from 'react';
import { motion } from 'framer-motion';

export interface IProps {
    height?: number;
    width?: number;
}

const LoadingIcon:FC<IProps> = ({ width, height }) => (
  <motion.div
    style={{
      width,
      height,
    }}
    initial={{ rotate: 0 }}
    animate={{ rotate: -360 }}
    transition={{
      type: 'tween',
      repeat: Infinity,
      repeatType: 'loop',
      duration: 0.8,
    }}
  >
    <motion.svg
      width={width}
      height={height}
      xmlns="http://www.w3.org/2000/svg"
      className="h-5 w-5"
      viewBox="0 0 20 20"
      fill="currentColor"
    >
      <path
        fillRule="evenodd"
        d="M4 2a1 1 0 011 1v2.101a7.002 7.002 0 0111.601 2.566 1 1 0 11-1.885.666A5.002 5.002 0 005.999 7H9a1 1 0 010 2H4a1 1 0 01-1-1V3a1 1 0 011-1zm.008 9.057a1 1 0 011.276.61A5.002 5.002 0 0014.001 13H11a1 1 0 110-2h5a1 1 0 011 1v5a1 1 0 11-2 0v-2.101a7.002 7.002 0 01-11.601-2.566 1 1 0 01.61-1.276z"
        clipRule="evenodd"
      />
    </motion.svg>
  </motion.div>

);

LoadingIcon.defaultProps = {
  height: 20,
  width: 20,
};

export default LoadingIcon;
