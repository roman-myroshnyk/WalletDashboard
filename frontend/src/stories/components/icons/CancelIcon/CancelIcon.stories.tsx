import { Meta, Story } from '@storybook/react/types-6-0';
import CancelIcon, { IProps } from '@/icons/CancelIcon';

export default {
  title: 'icons/CancelIcon',
  component: CancelIcon,
  argType: {
    height: {
      controls: 'number',
    },
    width: {
      controls: 'number',
    },
  },
} as Meta;

const Template:Story<IProps> = (args) => (
  <CancelIcon {...args} />
);

export const Default = Template.bind({});

Default.args = {
  height: 24,
  width: 24,
};
