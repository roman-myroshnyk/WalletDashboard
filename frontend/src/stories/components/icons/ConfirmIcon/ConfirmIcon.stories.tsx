import { Meta, Story } from '@storybook/react/types-6-0';
import ConfirmIcon, { IProps } from '@/icons/ConfirmIcon';

export default {
  title: 'icons/ConfirmIcon',
  component: ConfirmIcon,
  argType: {
    height: {
      controls: 'number',
    },
    width: {
      controls: 'number',
    },
  },
} as Meta;

const Template:Story<IProps> = (args) => (
  <ConfirmIcon {...args} />
);

export const Default = Template.bind({});

Default.args = {
  height: 24,
  width: 24,
};
