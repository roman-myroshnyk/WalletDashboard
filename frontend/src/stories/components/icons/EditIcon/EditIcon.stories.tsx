import { Meta, Story } from '@storybook/react/types-6-0';
import EditIcon, { IProps } from '@/icons/EditIcon';

export default {
  title: 'icons/EditIcon',
  component: EditIcon,
  argType: {
    height: {
      controls: 'number',
    },
    width: {
      controls: 'number',
    },
  },
} as Meta;

const Template:Story<IProps> = (args) => (
  <EditIcon {...args} />
);

export const Default = Template.bind({});

Default.args = {
  height: 24,
  width: 24,
};
