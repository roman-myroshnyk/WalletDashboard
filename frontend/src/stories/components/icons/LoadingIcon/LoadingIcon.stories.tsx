import { Meta, Story } from '@storybook/react/types-6-0';
import LoadingIcon, { IProps } from '@/icons/LoadingIcon';

export default {
  title: 'icons/LoadingIcon',
  component: LoadingIcon,
  argType: {
    height: {
      controls: 'number',
    },
    width: {
      control: 'number',
    },
  },
} as Meta;

const Template:Story<IProps> = (args) => (
  <LoadingIcon {...args} />
);

export const Default = Template.bind({});

Default.args = {
  height: 20,
  width: 20,
};
