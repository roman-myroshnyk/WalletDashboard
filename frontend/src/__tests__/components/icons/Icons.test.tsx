import { render } from '@testing-library/react';
import CancelIcon from '@/icons/CancelIcon';
import ConfirmIcon from '@/icons/ConfirmIcon';
import EditIcon from '@/icons/EditIcon';
import LoadingIcon from '@/icons/LoadingIcon';
import WarningIcon from '@/icons/WarningIcon';

describe('Icons', () => {
  it('renders CancelIcon', () => {
    const { container } = render(<CancelIcon />);
    expect(container).toMatchSnapshot();
  });

  it('renders ConfirmIcon', () => {
    const { container } = render(<ConfirmIcon />);
    expect(container).toMatchSnapshot();
  });

  it('renders EditIcon', () => {
    const { container } = render(<EditIcon />);
    expect(container).toMatchSnapshot();
  });

  it('renders LoadingIcon', () => {
    const { container } = render(<LoadingIcon />);
    expect(container).toMatchSnapshot();
  });

  it('renders WarningIcon', () => {
    const { container } = render(<WarningIcon />);
    expect(container).toMatchSnapshot();
  });
});
