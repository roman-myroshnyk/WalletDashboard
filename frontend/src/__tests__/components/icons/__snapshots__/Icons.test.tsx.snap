// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icons renders CancelIcon 1`] = `
<div>
  <svg
    aria-label="cancelIcon"
    class="h-6 w-6"
    data-testid="cancelIcon"
    fill="none"
    height="24"
    opacity="0"
    stroke="currentColor"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M6 18L18 6M6 12"
      pathLength="1"
      stroke-dasharray="0px 1px"
      stroke-dashoffset="0px"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
    <path
      d="M6 6M6 6l12 12"
      pathLength="1"
      stroke-dasharray="0px 1px"
      stroke-dashoffset="0px"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
</div>
`;

exports[`Icons renders ConfirmIcon 1`] = `
<div>
  <svg
    aria-label="confirmIcon"
    class="h-6 w-6"
    fill="none"
    height="24"
    opacity="0"
    stroke="currentColor"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M5 13l4 4L19 7"
      pathLength="1"
      stroke-dasharray="0px 1px"
      stroke-dashoffset="0px"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
</div>
`;

exports[`Icons renders EditIcon 1`] = `
<div>
  <svg
    aria-label="editIcon"
    class="h-6 w-6"
    fill="none"
    height="24"
    opacity="0"
    stroke="currentColor"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M11 5H6a2 2 0 00-2 2v11a2 2 0 002 2h11a2 2 0 002-2v-5m-1.414-9.414a2 2 0 112.828 2.828L11.828 15H9v-2.828l8.586-8.586z"
      pathLength="1"
      stroke-dasharray="0px 1px"
      stroke-dashoffset="0px"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
</div>
`;

exports[`Icons renders LoadingIcon 1`] = `
<div>
  <div
    style="width: 20px; height: 20px; transform: none;"
  >
    <svg
      aria-label="loadingIcon"
      class="h-5 w-5"
      fill="currentColor"
      height="20"
      viewBox="0 0 20 20"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M4 2a1 1 0 011 1v2.101a7.002 7.002 0 0111.601 2.566 1 1 0 11-1.885.666A5.002 5.002 0 005.999 7H9a1 1 0 010 2H4a1 1 0 01-1-1V3a1 1 0 011-1zm.008 9.057a1 1 0 011.276.61A5.002 5.002 0 0014.001 13H11a1 1 0 110-2h5a1 1 0 011 1v5a1 1 0 11-2 0v-2.101a7.002 7.002 0 01-11.601-2.566 1 1 0 01.61-1.276z"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;

exports[`Icons renders WarningIcon 1`] = `
<div>
  <div
    style="display: block; align-items: center; justify-content: center; width: 24px; height: 24px; transform: scale(0) translateZ(0);"
  >
    <svg
      aria-label="warningIcon"
      class="h-5 w-5"
      fill="currentColor"
      height="24"
      viewBox="0 0 20 20"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M8.257 3.099c.765-1.36 2.722-1.36 3.486 0l5.58 9.92c.75 1.334-.213 2.98-1.742 2.98H4.42c-1.53 0-2.493-1.646-1.743-2.98l5.58-9.92zM11 13a1 1 0 11-2 0 1 1 0 012 0zm-1-8a1 1 0 00-1 1v3a1 1 0 002 0V6a1 1 0 00-1-1z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;
